silk-react-components/src/HierarchicalMapping/Components/MappingRule/Forms/ObjectMappingRuleForm.jsx

```diff
@@ -14,291 +15,213 @@ import {
-                    .onNext({id});
+                EventEmitter.emit(MESSAGES.RULE_VIEW.UNCHANGED, { id });
             }
         }
-
         this.setState({
-            [name]: value,
-            changed: touched,
+            modifiedValues: newModifiedValues,
+            changed,
         });
-    },
+    }
+
+    /**
+     * handle form close event
+     * @param event
+     */
     handleClose(event) {
         event.stopPropagation();
-        const id = _.get(this.props, 'id', 0);
-        hierarchicalMappingChannel.subject('ruleView.unchanged').onNext({id});
-        hierarchicalMappingChannel.subject('ruleView.close').onNext({id});
-    },
-    getExampleView() {
-        if (this.state.pattern) {
-            return (
-                <ExampleView
-                    id={this.props.parentId || 'root'}
-                    rawRule={{
-                        type: MAPPING_RULE_TYPE_URI,
-                        pattern: this.state.pattern,
-                    }}
-                    ruleType={MAPPING_RULE_TYPE_URI}
-                />
-            );
-        } else if (this.state.uriRule) {
-            return (
-                <ExampleView
-                    id={this.props.parentId || 'root'}
-                    rawRule={this.state.uriRule}
-                    ruleType={this.state.uriRule.type}
-                />
-            );
-        }
+        const { id = 0 } = this.props;
+        EventEmitter.emit(MESSAGES.RULE_VIEW.UNCHANGED, { id });
+        EventEmitter.emit(MESSAGES.RULE_VIEW.CLOSE, { id });
+    }
 
-        return false;
-    },
-    // template rendering
+    /**
+     * React's lifecycle method
+     * @returns {*}
+     */
     render() {
-        const {id, parentId} = this.props;
+        const { id, parentId, parent } = this.props;
+        const {
+            saveObjectError, loading, modifiedValues, changed,
+        } = this.state;
 
         const autoCompleteRuleId = id || parentId;
 
-        const {error} = this.state;
-
-        const type = this.state.type;
-
-        if (this.state.loading) {
+        if (loading) {
             return <Spinner />;
         }
 
         // FIXME: also check if data really has changed before allow saving
         const allowConfirm =
-            type === MAPPING_RULE_TYPE_ROOT
-                ? true
-                : !_.isEmpty(this.state.targetProperty);
-
-        const errorMessage = error ? (
-            <ErrorView {...error.response.body} />
-        ) : (
-            false
+            modifiedValues.type === MAPPING_RULE_TYPE_ROOT || !_.isEmpty(modifiedValues.targetProperty);
+        const errorMessage = saveObjectError && (
+            <ErrorView {...saveObjectError} />
         );
 
-        const title =
-            // TODO: add source path if: parent, not edit, not root element
-            !id ? <CardTitle>Add object mapping</CardTitle> : false;
+        // TODO: add source path if: parent, not edit, not root element
```

Comment:
Are you aware of the TODO / is this something which can be fixed now?

Reply:
Rly, I don't understand what's mean 'source path'? maybe this is the feature?